packages/tx/src/features/util.ts

```diff
@@ -0,0 +1,138 @@
+import { Common, Mainnet } from '@ethereumjs/common'
+import {
+  Address,
+  MAX_INTEGER,
+  MAX_UINT64,
+  bigIntToHex,
+  bytesToBigInt,
+  bytesToHex,
+  toBytes,
+} from '@ethereumjs/util'
+
+import { paramsTx } from '../params.js'
+import { checkMaxInitCodeSize, validateNotArray } from '../util.js'
+
+import type { TransactionInterface, TransactionType, TxData, TxOptions } from '../types.js'
+
+export function getCommon(common?: Common): Common {
+  return common?.copy() ?? new Common({ chain: Mainnet })
+}
+
+/**
+ * Validates that an object with BigInt values cannot exceed the specified bit limit.
+ * @param values Object containing string keys and BigInt values
+ * @param bits Number of bits to check (64 or 256)
+ * @param cannotEqual Pass true if the number also cannot equal one less the maximum value
+ */
+export function valueBoundaryCheck( // TODO: better method name
+  values: { [key: string]: bigint | undefined },
+  bits = 256,
+  cannotEqual = false,
+) {
+  for (const [key, value] of Object.entries(values)) {
+    switch (bits) {
+      case 64:
+        if (cannotEqual) {
+          if (value !== undefined && value >= MAX_UINT64) {
+            // TODO: error msgs got raised to a error string handler first, now throws "generic" error
+            throw new Error(`${key} cannot equal or exceed MAX_UINT64 (2^64-1), given ${value}`)
+          }
+        } else {
+          if (value !== undefined && value > MAX_UINT64) {
+            throw new Error(`${key} cannot exceed MAX_UINT64 (2^64-1), given ${value}`)
+          }
+        }
+        break
+      case 256:
+        if (cannotEqual) {
+          if (value !== undefined && value >= MAX_INTEGER) {
+            throw new Error(`${key} cannot equal or exceed MAX_INTEGER (2^256-1), given ${value}`)
+          }
+        } else {
+          if (value !== undefined && value > MAX_INTEGER) {
+            throw new Error(`${key} cannot exceed MAX_INTEGER (2^256-1), given ${value}`)
+          }
+        }
+        break
+      default: {
+        throw new Error('unimplemented bits value')
+      }
+    }
+  }
+}
+
+type Mutable<T> = {
+  -readonly [P in keyof T]: T[P]
+}
+
+// This is (temp) a shared method which reflects `super` logic which were called from all txs and thus
+// represents the constructor of baseTransaction
+// Note: have to use `Mutable` to write to readonly props. Only call this in constructor of txs.
+export function sharedConstructor(
+  tx: Mutable<TransactionInterface>,
+  txData: TxData[TransactionType],
+  opts: TxOptions = {},
+) {
+  // LOAD base tx super({ ...txData, type: TransactionType.Legacy }, opts)
+  tx.common = getCommon(opts.common)
+  tx.common.updateParams(opts.params ?? paramsTx)
+
+  validateNotArray(txData) // is this necessary?
```

Comment:
As I recall, this got added because there was some weird edge case where we were accepting `txData` where somehow an array got passed in.